tests/test_codex_feed_hooks.py

```
@@ -420,6 +420,138 @@ def test_codex_monitor_exits_when_workspace_has_no_surfaces(cli_path: str, root:
+            env=env,
+        )
+        try:
+            wait_for_monitor_pids(session_id, present=True, timeout=5)
+            deadline = time.time() + 5
+            while time.time() < deadline:
+                if monitor.poll() is not None:
+                    stdout, stderr = monitor.communicate(timeout=1)
+                    raise AssertionError(
+                        "monitor exited before owner PID was terminated; "
+                        f"stdout={stdout}\nstderr={stderr}"
+                    )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)